torch/_dynamo/config.py

```diff
@@ -374,6 +374,10 @@ def default_debug_dir_root():
 # used for testing
 inject_BUILD_SET_unimplemented_TESTING_ONLY = False
 
+# How many iterations a for loop has to be before we decide to automatically
+# convert it to a function body.
+for_loop_medium_size_boundary = 100
+
```

Comment:
should have a config option to disable this feature. Since it seems to test well we should start with a default of False in this PR and then flip it to True in a second PR.